diff --git a/dist/cjs/index-24665584.cjs b/dist/cjs/index-24665584.cjs
index 4f3b34eb3c85a3337f61ee7b21b029da1b9a4123..da5c6008e08add51da22413501edfafbf7c28dbf 100644
--- a/dist/cjs/index-24665584.cjs
+++ b/dist/cjs/index-24665584.cjs
@@ -10410,6 +10410,15 @@ function computeRealizedPriceImpact(trade) {
   return trade.priceImpact.subtract(realizedLpFeePercent);
 }
 function getPriceImpactWarning(priceImpact) {
+  if (
+    !priceImpact ||
+    priceImpact.toSignificant() === 'NaN' ||
+    priceImpact.toSignificant() === 'Infinity' ||
+    !priceImpact.greaterThan
+  ) {
+    return 'error';
+  }
+
   if (priceImpact.greaterThan(ALLOWED_PRICE_IMPACT_HIGH)) return 'error';
   if (priceImpact.greaterThan(ALLOWED_PRICE_IMPACT_MEDIUM)) return 'warning';
   return;
@@ -10554,6 +10563,22 @@ function useUSDCValue(currencyAmount) {
 function usePriceImpact(trade) {
   return React.useMemo(function () {
     var marketPriceImpact = trade ? computeRealizedPriceImpact(trade) : undefined;
+
+    if (
+      marketPriceImpact &&
+      (
+        marketPriceImpact.toSignificant() === 'NaN' ||
+        marketPriceImpact.toSignificant() === 'Infinity' ||
+        // Check for negative price impact
+        Number(marketPriceImpact.toSignificant()) < 0
+      )
+    ) {
+      return {
+        percent: marketPriceImpact,
+        warning: 'error'
+      };
+    }
+
     return marketPriceImpact ? {
       percent: marketPriceImpact,
       warning: getPriceImpactWarning(marketPriceImpact)
diff --git a/dist/index-53ec290b.js b/dist/index-53ec290b.js
index e7da428525171db63ee79a0b73e18aeb3bcad3b4..f4eff4b27f503fe2f4caf32f985b3759c0e8b339 100644
--- a/dist/index-53ec290b.js
+++ b/dist/index-53ec290b.js
@@ -10359,6 +10359,15 @@ function computeRealizedPriceImpact(trade) {
   return trade.priceImpact.subtract(realizedLpFeePercent);
 }
 function getPriceImpactWarning(priceImpact) {
+  if (
+    !priceImpact ||
+    priceImpact.toSignificant() === 'NaN' ||
+    priceImpact.toSignificant() === 'Infinity' ||
+    !priceImpact.greaterThan
+  ) {
+    return 'error';
+  }
+
   if (priceImpact.greaterThan(ALLOWED_PRICE_IMPACT_HIGH)) return 'error';
   if (priceImpact.greaterThan(ALLOWED_PRICE_IMPACT_MEDIUM)) return 'warning';
   return;
@@ -10501,8 +10510,24 @@ function useUSDCValue(currencyAmount) {
 }
 
 function usePriceImpact(trade) {
-  return useMemo(function () {
+  return React.useMemo(function () {
     var marketPriceImpact = trade ? computeRealizedPriceImpact(trade) : undefined;
+
+    if (
+      marketPriceImpact &&
+      (
+        marketPriceImpact.toSignificant() === 'NaN' ||
+        marketPriceImpact.toSignificant() === 'Infinity' ||
+        // Check for negative price impact
+        Number(marketPriceImpact.toSignificant()) < 0
+      )
+    ) {
+      return {
+        percent: marketPriceImpact,
+        warning: 'error'
+      };
+    }
+
     return marketPriceImpact ? {
       percent: marketPriceImpact,
       warning: getPriceImpactWarning(marketPriceImpact)
